lib/scout_apm/framework_integrations/rails_3_or_4.rb

```diff
@@ -71,7 +71,11 @@ def database_engine
       #
       # We avoid this issue by not calling .respond_to? here, and instead using the less optimal `rescue nil` approach
       def raw_database_adapter
-        adapter = ActiveRecord::Base.connection_config[:adapter].to_s rescue nil
+        adapter = ActiveRecord::Base.connection_db_config.configuration_hash[:adapter] rescue nil
+
+        if adapter.nil?
+          adapter = ActiveRecord::Base.connection_config[:adapter].to_s rescue nil
+        end
 
         if adapter.nil?
           adapter = ActiveRecord::Base.configurations[env]["adapter"]
```

Comment:
Here is another depreciation, using `[]` for the `configurations` method. If it is your intention please consider replacing it with `configs_for`
```ruby
adapter = ActiveRecord::Base.configurations.configs_for(env)["adapter"]
```
https://github.com/rails/rails/pull/33637